In [1]:
from bs4 import BeautifulSoup as bs
from splinter import Browser
import time
import datetime
import pandas as pd
import lxml
import re

In [2]:
df = pd.read_csv("./04-19-23.csv")
df

id                          title     subtitle  \
0      1            Multiples of 3 or 5    Problem 1   
1      2         Even Fibonacci numbers    Problem 2   
2      3           Largest prime factor    Problem 3   
3      4     Largest palindrome product    Problem 4   
4      5              Smallest multiple    Problem 5   
..   ...                            ...          ...   
824  825                   Chasing Game  Problem 825   
825  826                Birds on a Wire  Problem 826   
826  827  Pythagorean Triple Occurrence  Problem 827   
827  828              Numbers Challenge  Problem 828   
828  829                Integral Fusion  Problem 829   

                                               content  \
0    \nIf we list all the natural numbers below 10 ...   
1    \nEach new term in the Fibonacci sequence is g...   
2    \nThe prime factors of 13195 are 5, 7, 13 and ...   
3    \nA palindromic number reads the same both way...   
4    \n2520 is the smallest number that can be divi...   
..                                                 ...   
824  \nTwo cars are on a circular track of total le...   
825  \nConsider a wire of length 1 unit between two...   
826  \n\nDefine Q(n) to be the smallest number that...   
827  \nIt is a common recreational problem to make ...   
828  \nGiven any integer n>1 a binary factor tree T...   

                                          html_content         release_date  \
0    <p>If we list all the natural numbers below 10...  2001-10-05T18:00:00   
1    <p>Each new term in the Fibonacci sequence is ...  2001-10-19T18:00:00   
2    <p>The prime factors of 13195 are 5, 7, 13 and...  2001-11-02T18:00:00   
3    \n<p>A palindromic number reads the same both ...  2001-11-16T18:00:00   
4    <p>2520 is the smallest number that can be div...  2001-11-30T18:00:00   
..                                                 ...                  ...   
824  <p>Two cars are on a circular track of total l...  2023-01-15T07:00:00   
825  <p>Consider a wire of length 1 unit between tw...  2023-01-22T10:00:00   
826  <p>\nDefine $Q(n)$ to be the smallest number t...  2023-01-28T13:00:00   
827  <p>It is a common recreational problem to make...  2023-02-04T16:00:00   
828  <p>Given any integer $n&gt;1$ a <i>binary fact...  2023-02-11T19:00:00   

     solved_by_count  difficulty  
0             970022           5  
1             774310           5  
2             557844           5  
3             493961           5  
4             497288           5  
..               ...         ...  
824              117          60  
825              184          45  
826              144          50  
827              392          15  
828              172          45  

[829 rows x 8 columns]

In [3]:
df["content"][56]

'\nIt is possible to show that the square root of two can be expressed as an infinite continued fraction.\n2=1+12+12+12+…\nBy expanding this for the first four iterations, we get:\n1+12=32=1.5\n1+12+12=75=1.4\n1+12+12+12=1712=1.41666…\n1+12+12+12+12=4129=1.41379…\nThe next three expansions are 9970, 239169, and 577408, but the eighth expansion, 1393985, is the first example where the number of digits in the numerator exceeds the number of digits in the denominator.\nIn the first one-thousand expansions, how many fractions contain a numerator with more digits than the denominator?\n'

In [4]:
# use HTML content as well as text content as columns (html content can be rendered correctly with MathJax in a markdown cell)
# problem 57 example

<p>It is possible to show that the square root of two can be expressed as an infinite continued fraction.</p>
<p class="center">$\sqrt 2 =1+ \frac 1 {2+ \frac 1 {2 +\frac 1 {2+ \dots}}}$</p>
<p>By expanding this for the first four iterations, we get:</p>
<p>$1 + \frac 1 2 = \frac  32 = 1.5$<br />
$1 + \frac 1 {2 + \frac 1 2} = \frac 7 5 = 1.4$<br />
$1 + \frac 1 {2 + \frac 1 {2+\frac 1 2}} = \frac {17}{12} = 1.41666 \dots$<br />
$1 + \frac 1 {2 + \frac 1 {2+\frac 1 {2+\frac 1 2}}} = \frac {41}{29} = 1.41379 \dots$<br /></p>
<p>The next three expansions are $\frac {99}{70}$, $\frac {239}{169}$, and $\frac {577}{408}$, but the eighth expansion, $\frac {1393}{985}$, is the first example where the number of digits in the numerator exceeds the number of digits in the denominator.</p>
<p>In the first one-thousand expansions, how many fractions contain a numerator with more digits than the denominator?</p>

In [ ]:
# problems per difficulty level
# problem solvers count buckets
# word map from problem statements
# release date chart?